### PR TITLE
Publish 3.5.4 to docker registry

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
Publish 3.5.4 to docker registry

### Why are the changes needed?
3.5.4 release step

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
CI

### CODE REVIEW

**Comment by LuciferYang** [comment]
will re-triggered after the tar package is released

**Comment by LuciferYang** [comment]
image

still
```
132.9 failed: Connection timed out.
132.9 failed: Network is unreachable.
```

**Comment by dongjoon-hyun** [comment]
BTW, ASF Archive is ready now. Could you re-trigger once more?

https://archive.apache.org/dist/spark/spark-3.5.4/

Screenshot 2024-12-20 at 09 05 40

**Comment by dongjoon-hyun** [comment]
It's working now.
```
$ curl -LO https://archive.apache.org/dist/spark/spark-3.5.4/spark-3.5.4-bin-hadoop3.tgz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  382M  100  382M    0     0  12.9M      0  0:00:29  0:00:29 --:--:-- 12.0M
```

**Comment by LuciferYang** [comment]
image


Maybe it's the same issue as https://github.com/apache/spark/pull/48406 ？

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -257,6 +257,11 @@ jobs:
           path: ~/.cache/coursier
           key: build-${{ inputs.spark }}-scala${{ inputs.scala }}-java${{ inputs.java }}-coursier
 
+      - name: Install R
```
Comment (LuciferYang) [inline .github/workflows/main.yml:260]: Perhaps this should be done in a preceding pr but let me try it out first.

Comment (LuciferYang) [reply .github/workflows/main.yml:260]: @dongjoon-hyun It seems that we need to manually install the R environment for ubuntu-latest, after the installation, the tests for 3.5.4 all seem to have succeeded.

Before

https://github.com/apache/spark-docker/actions/runs/12431196505/job/34721415046

After

https://github.com/apache/spark-docker/actions/runs/12436092928/job/34723236153?pr=77

Comment (LuciferYang) [reply .github/workflows/main.yml:260]: @dongjoon-hyun 

Is this solution acceptable? I referenced https://github.com/apache/spark/pull/48406. 

Should I submit a separate pr or can it be completed in this one?

also cc @Yikun @yaooqinn

Comment (LuciferYang) [reply .github/workflows/main.yml:260]: I make a separate one : https://github.com/apache/spark-docker/pull/78

**Review by HyukjinKwon — APPROVED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (LuciferYang) [reply .github/workflows/main.yml:260]: > I make a separate one : #78

Let me rebase this one

**Comment by LuciferYang** [comment]
all test passed, merged into master

**Comment by LuciferYang** [comment]
It looks like the publish to `apache/spark` was successful:
- https://github.com/apache/spark-docker/actions/runs/12451615569
- https://hub.docker.com/r/apache/spark/tags

cc @HyukjinKwon  @Yikun @dongjoon-hyun @yaooqinn

**Comment by dongjoon-hyun** [comment]
Thank you, @LuciferYang !

**Pull request merged**